server/src/main/java/org/opensearch/index/codec/CodecAliases.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec;
+
+import org.apache.lucene.codecs.Codec;
+
+import java.util.Set;
+
+/**
+ * This {@link CodecAliases} allows us to manage the settings with {@link Codec}.
+ *
+ * @opensearch.internal
+ */
+public interface CodecAliases {
+    Set<String> aliases();
```

Comment:
Please add javadocs. no `null` should be returned

Reply:
@reta - Would javadocs suffice here? Unless there is a mechanism to validate it is not null during compile time, there is a possibility the value could be null.

Reply:
> @reta - Would javadocs suffice here?

Yeah yeah, just suggested javadoc to mention that explicitly

Reply:
thanks folks, will add this